pyworkflow/protocol/protocol.py

```diff
@@ -734,7 +735,10 @@ def hasLinkedInputs(self):
                 else:
                     emptyPointers.append(paramName)
 
-        return linkedPointers and not emptyPointers
+        if not self.worksInStreaming() and isinstance(obj, Set) and obj.isStreamOpen():
```

Comment:
This will work only for the last item in the loop. Shouldn't we resolve the status inside the loop?